DocBlocks/src/Terminator.php

```diff
@@ -0,0 +1,38 @@
+<?php declare(strict_types = 1);
+
+class Terminator
+{
+	/**
+	 * @var IVisualProcessor
+	 */
+	private $targetProcessor;
+
+	/**
+	 * @var IGun
+	 */
+	private $gun;
+
+
+	public function __construct(IVisualProcessor $targetProcessor, GunProvider $gunProvider)
+	{
+		$this->targetProcessor = $targetProcessor;
+		$this->gun = $gunProvider->acquireGun();
+	}
+
+
+	/**
+	 * This is Terminator's purpose - to seek and destroy defined targets with final state of void (no targets left).
+	 *
+	 * @return void
```

Comment:
Pokud bychom používali nástroj, který by generoval api, tak by měl umět mergovat komentáře a typy z jazyka. Abychom neduplikovali kód do komentářů jen kvůli tomuto.

U komentářů v metodám bych měl jiný příklad - parent deklaruje base typ, ale implementace dostává konkrétnější. Potom si ji správně oanotuji, IDE to takto chápe a napovídá mi správný typ. Sice mě zase upozorňuje, že mi nesedí typy v komentáři oproti implementaci, ale komentáře nemají v PHP žádný sémantický význam, takže je možné si upozornění vypnout. Osobně do komentářů dávám jen to, co není možné podchytit jazykem.